docs/tutorials/rerank/rerank_pipeline_with_Amazon_Rerank_model_on_Amazon_Bedrock.md

```diff
@@ -0,0 +1,622 @@
+        "Washington, D.C. (also known as simply Washington or D.C., and officially as the District of Columbia) is the capital of the United States. It is a federal district.",
+        "Capital punishment (the death penalty) has existed in the United States since beforethe United States was a country. As of 2017, capital punishment is legal in 30 of the 50 states."
+    ]
+})
+
+response = bedrock_runtime_client.invoke_model(
+    modelId=modelId,
+    contentType=contentType,
+    accept=accept, 
+    body=body
+)
+results = json.loads(response.get('body').read())["results"]
+print(json.dumps(sorted(results, key=lambda x: x['index']),indent=2))
+```
+
+The reranking results are as follows:
+
+```
+[
+  {
+    "index": 0,
+    "relevance_score": 0.0025114635138098534
+  },
+  {
+    "index": 1,
+    "relevance_score": 2.4876490010363496e-05
+  },
+  {
+    "index": 2,
+    "relevance_score": 0.7711548724998493
+  },
+  {
+    "index": 3,
+    "relevance_score": 6.339210403977635e-06
+  }
+]
+```
+
+## 1. Create a connector and register the model
+
+To create a connector for the model, send the following request. If you are using self-managed OpenSearch, supply your AWS credentials:
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "Amazon Bedrock cross-encoder model",
+  "description": "Test connector for Amazon Bedrock cross-encoder model",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "credential": {
+    "access_key": "your_access_key",
+    "secret_key": "your_secret_key",
+    "session_token": "your_session_token"
+  },
+  "parameters": {
+    "region": "your_bedrock_model_region_like_us-west-2",
+    "service_name": "bedrock"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://bedrock-runtime.${parameters.region}.amazonaws.com/model/amazon.rerank-v1:0/invoke",
+      "headers": {
+        "x-amz-content-sha256": "required",
+        "content-type": "application/json"
+      },
+      "pre_process_function": """
+        def query_text = params.query_text;
+        def text_docs = params.text_docs;
+        def textDocsBuilder = new StringBuilder('[');
+        for (int i=0; i<text_docs.length; i++) {
+          textDocsBuilder.append('\"');
+          textDocsBuilder.append(text_docs[i]);
+          textDocsBuilder.append('\"');
+          if (i<text_docs.length - 1) {
+            textDocsBuilder.append(',');
+          }
+        }
+        textDocsBuilder.append(']');
+        def parameters = '{ \"query\": \"' + query_text + '\",  \"documents\": ' + textDocsBuilder.toString() + ' }';
+        return  '{\"parameters\": ' + parameters + '}';
+      """,
+      "request_body": "{ \"query\": \"${parameters.query}\", \"documents\": ${parameters.documents} }",
+      "post_process_function": """
+        if (params.result == null || params.result.length > 0) {
```

Comment:
```suggestion
        if (params.results == null || params.results.length == 0) {
```

Reply:
I'm going to remove unnecessary escape character and test again on my side.

Reply:
fixed on 9482b472b2bc68bba61267c87f484967362e2c5b